src/components/pages/register/register-page/components/register-form/RegisterForm.tsx

```diff
@@ -19,10 +22,10 @@ import AuthService from '@/lib/services/auth';
 import StorageUtil from '@/lib/utils/StorageUtil';
 
 import { initialValues } from './constants';
+import * as stylesMUI from './RegisterForm.styles';
 import { validationSchema } from './validation';
 
-import styles from '../left-block/LeftBlock.module.scss';
-
+import styles from './FormStyles.module.scss';
 const RegisterForm: FC<GetAllResponse> = ({ groups }) => {
   const router = useRouter();
   const toast = useToast();
```

Comment:
handleSubmit looks a bit cramped up. Could you split it up into smaller functions and maybe use a switch? First of all separate the registration logic and error handling logic into separate functions. It would make code more readable. Thanks!